### PR TITLE
Add GitHub actions in preperation for deprecating Jenkins

### PR DESCRIPTION
**Pull request by elichai**
Added a new script `test.sh` that runs everything we want, so it can be easy to run locally.
you can pass `-v` to the script to get verbose outputs. (we might want to enable that in the CI?)
Added github actions that run this script on Ubuntu, OSX and windows.
This is a continuation of #1130

~The Github Action already uses go 1.15, but haven't changed the docker file yet~
It will keep using Go 1.14 for now

### CODE REVIEW

**Comment by elichai** [comment]
So this is missing 1 thing:
The ability to re-run whenever the base branch changes, we can either achieve this by writing a custom action that re-triggers the PRs, or by writing an action that comments on all open PRs and then making that comment trigger it

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Increase page size on windows
+        if: runner.os == 'Windows'
+        shell: powershell
+        run: powershell -command .\.github\workflows\SetPageFileSize.ps1
+
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
```
Comment (svarogg) [inline .github/workflows/go.yml:37]: Currently we are still at version 1.14
Upgrade to 1.15 should happen for everything at the same time.

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/sh -ex
```
Comment (svarogg) [inline test.sh:1]: This does much more then just test, so I'd rather it be called that accordingly.

**Review by svarogg — CHANGES_REQUESTED** [review]

File: test.sh (hunk 2 above)
Comment (elichai) [reply test.sh:1]: I consider everything in it part of our "tests" but if you have a better name I'm all ears :)

File: .github/workflows/go.yml (hunk 1 above)
Comment (elichai) [reply .github/workflows/go.yml]: Done

**Review by svarogg — APPROVED** [review]

**Pull request merged**